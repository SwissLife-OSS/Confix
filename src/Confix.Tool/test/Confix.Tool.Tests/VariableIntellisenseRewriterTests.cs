using System.Text.Json.Nodes;
using Confix.Tool.Schema;
using Json.More;

namespace ConfiX.Entities.Component.Configuration;

public class VariableIntellisenseRewriterTests
{
    [Fact]
    public void Rewrite_JsonObjectOfString_CorrectlyRewrites()
    {
        // arrange
        var obj = new JsonObject
        {
            ["type"] = "string",
            ["hasVariable"] = true,
        };
        var context = new VariableIntellisenseContext(
          new JsonObject()
        );
        var sut = new VariableIntellisenseRewriter();

        // act
        var result = sut.Rewrite(obj, context);

        // assert
        Assert.True(new JsonObject
        {
            ["anyOf"] = new JsonArray{
              context.VariableReference,
              obj,
            }
        }.IsEquivalentTo(result));
    }
}