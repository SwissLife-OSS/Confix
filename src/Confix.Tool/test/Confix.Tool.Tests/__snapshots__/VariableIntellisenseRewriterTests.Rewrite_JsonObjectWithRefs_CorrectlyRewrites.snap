{
  "$id": "https://example.com/schemas/customer",
  "type": "object",
  "properties": {
    "first_name": {
      "$ref": "#/$defs/name"
    },
    "last_name": {
      "$ref": "#/$defs/name"
    },
    "shipping_address": {
      "$ref": "/schemas/address"
    },
    "billing_address": {
      "$ref": "/schemas/address"
    }
  },
  "required": [
    "first_name",
    "last_name",
    "shipping_address",
    "billing_address"
  ],
  "$defs": {
    "name": {
      "anyOf": [
        {
          "$ref": "#/definitions/variable"
        },
        {
          "type": "string"
        }
      ]
    }
  }
}
